src/common/Header/index.js

```diff
@@ -15,6 +20,26 @@ import camera from "./icons/camera.svg";
 import search from "./icons/search.svg";
 
 export const Header = () => {
+
+  const dispatch = useDispatch();
+  const location = useLocation();
+
+  const query = useQueryParameter("search");
+  const replaceQueryParam = useReplaceQueryParameter();
+
+  const onInputChange = ({ target }) => {
+    replaceQueryParam({
+      key: "search",
+      value: target.value.trim(),
+    });
+  };
+
+  const onFormSubmit = (event) => {
```

Comment:
jak coś to zwróć uwagę koniecznie jaka jest teraz forma Searcha, bo został jedynie Input, bez formularza, takze zeby nie pomieszać tych wszystkich wartości :)